mjpeg/__init__.py

```diff
@@ -83,6 +93,28 @@ def read_data(buf, stream, length):
     return buf
 
 
+async def aread_data(buf, stream: aiohttp.BodyPartReader, length: int):
+    chunk_size = 8192
+    if chunk_size <= length:
+        while chunk_size > length and chunk_size != 1:
+            chunk_size = chunk_size // 2
```

Comment:
If I read the above code correctly, you are trying to find a chunk_size that is a power of two and that is not bigger than length? If that is the case, shouldn't the boolean expression in the if condition read `chunk_size > length`?

The way you have it written now, the body of the if statement is only executed if `chunk_size <= length`, but then the while loop terminates immediately without any iterations because `chunk_size > length` will always evaluate to False.

Reply:
Another silly mistake and another good catch!

I'm honestly not incredibly satisfied with this logic to begin with.  It would be much cleaner if `aiohttp.BodyPartReader` had a buffer interface that would support `readinto()`.

I'm open to suggestions on making it cleaner and more efficient

Reply:
Are you sure this is the correct fix? The way I read it now, if chunk_size is less than or equal to length, it will decrease chunk_size until it is one. Is that what you intended?